src/Cli/Microsoft.DotNet.Cli.Utils/RuntimeEnvironment.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal enum Platform
+    {
+        Unknown = 0,
+        Windows = 1,
+        Linux = 2,
+        Darwin = 3,
+        FreeBSD = 4
+    }
+
+    internal static class RuntimeEnvironment
+    {
+        private static readonly Lazy<Platform> _platform = new Lazy<Platform>(DetermineOSPlatform);
+        private static readonly Lazy<DistroInfo> _distroInfo = new Lazy<DistroInfo>(LoadDistroInfo);
+
+        public static Platform OperatingSystemPlatform { get; } = GetOSPlatform();
+        public static string OperatingSystemVersion { get; } = GetOSVersion();
+        public static string OperatingSystem { get; } = GetOSName();
+
+        // toolset-tasks.csproj needs to build for full .NET Framework, and needs to get the current
+        // RuntimeIdentifier of the machine. Because of this, it needs to implement
+        // it's own GetRuntimeIdentifier(). All other places should use RuntimeInformation.RuntimeIdentifier.
+#if TOOLSET_TASKS
+        private static readonly string OverrideEnvironmentVariableName = "DOTNET_RUNTIME_ID";
+
+        public static string GetRuntimeIdentifier()
```

Comment:
It looks like we're not even using the [value that is returned](https://github.com/dotnet/sdk/blob/master/src/Layout/redist/targets/GetRuntimeInformation.targets#L6) in the `Rid` output parameter of the GetCurrentRuntimeInformation task.  So it looks like we actually can delete this implementation of `GetRuntimeIdentifier`.

Reply:
Nice catch! After seeing that isn't used, I was able to clean up a lot more code as well.